contracts/ModifiedTokenDistro.sol

```diff
@@ -0,0 +1,359 @@
+        //     "TokenDistro::changeAddress: ADDRESS_ALREADY_IN_USE"
+        // );
+
+        require(
+            !hasRole(DISTRIBUTOR_ROLE, msg.sender) && !hasRole(DISTRIBUTOR_ROLE, newAddress),
+            'TokenDistro::changeAddress: DISTRIBUTOR_ROLE_NOT_A_VALID_ADDRESS'
+        );
+        // balance adds instead of overwrites
+        balances[newAddress].allocatedTokens += balances[msg.sender].allocatedTokens;
+        balances[msg.sender].allocatedTokens = 0;
+
+        balances[newAddress].claimed += balances[msg.sender].claimed;
+        balances[msg.sender].claimed = 0;
+
+        emit ChangeAddress(msg.sender, newAddress);
+    }
+
+    /**
+     * Function to get the current timestamp from the block
+     */
+    function getTimestamp() public view virtual override returns (uint256) {
+        return block.timestamp;
+    }
+
+    /**
+     * Function to get the total claimable tokens at some moment
+     * @param timestamp Unix time to check the number of tokens claimable
+     * @return Number of tokens claimable at that timestamp
+     */
+    function globallyClaimableAt(uint256 timestamp) public view override returns (uint256) {
+        if (timestamp < startTime) return 0;
+        if (timestamp < cliffTime) return initialAmount;
+        if (timestamp > startTime + duration) return totalTokens;
+
+        uint256 deltaTime = timestamp - startTime;
+        return initialAmount + (deltaTime * lockedAmount) / duration;
+    }
+
+    /**
+     * Function to get the unlocked tokes at some moment for a specific address
+     * @param recipient account to query
+     * @param timestamp Instant of time in which the calculation is made
+     */
+    function claimableAt(address recipient, uint256 timestamp) public view override returns (uint256) {
+        require(!hasRole(DISTRIBUTOR_ROLE, recipient), 'TokenDistro::claimableAt: DISTRIBUTOR_ROLE_CANNOT_CLAIM');
+        require(timestamp >= getTimestamp(), 'TokenDistro::claimableAt: NOT_VALID_PAST_TIMESTAMP');
+        uint256 unlockedAmount = (globallyClaimableAt(timestamp) * balances[recipient].allocatedTokens) / totalTokens;
+
+        return unlockedAmount - balances[recipient].claimed;
+    }
+
+    /**
+     * Function to get the unlocked tokens for a specific address. It uses the current timestamp
+     * @param recipient account to query
+     */
+    function claimableNow(address recipient) public view override returns (uint256) {
+        return claimableAt(recipient, getTimestamp());
+    }
+
+    /**
+     * Function that allows the DEFAULT_ADMIN_ROLE to change a recipient in case it wants to cancel an allocation
+     * @dev The change can only be made when cancelable is true and to an address that has not previously received
+     * an allocation and the distributor cannot change its address
+     *
+     * Emits a {ChangeAddress} event.
+     *
+     */
+    function transferAllocation(address prevRecipient, address newRecipient) external override {
```

Comment:
I noticed you renamed the function along with removing this check
```solidity
        require(cancelable, 'TokenDistro::cancelAllocation: NOT_CANCELABLE');
```
This change will make the `cancelable` property unused, and we cannot remove it unless we fill it with a dummy variable for memory safety reasons.
Do you still think it's better to rename it and remove the check?

Reply:
I added back the check - Do you think there will be consequences for renaming the function for memory safety?

Reply:
Renaming might result in fallback call when function is invoked. @sembrestels would appreciate your comment on this.

Reply:
Please add this comment above the function definition

// this function should only be called manually